ethdb/privateapi/ethbackend.go

```diff
@@ -516,6 +516,22 @@ func (s *EthBackendServer) EngineGetPayloadV2(ctx context.Context, req *remote.E
 	return &types2.ExecutionPayloadV2{Payload: payload, Withdrawals: withdrawals}, nil
 }
 
+func (s *EthBackendServer) EngineGetPayloadV3(ctx context.Context, req *remote.EngineGetPayloadRequest) (*types2.ExecutionPayloadV3, error) {
+	block, payload, err := s.engineGetPayload(req)
+	if err != nil {
+		return nil, err
+	}
+	withdrawals := ConvertWithdrawalsToRpc(block.Withdrawals())
+
+	var excessDataGasReply *types2.H256
+	if block.ExcessDataGas() != nil {
+		var excessDataGas uint256.Int
+		excessDataGas.SetFromBig(block.Header().BaseFee)
+		excessDataGasReply = gointerfaces.ConvertUint256IntToH256(&excessDataGas)
+	}
+	return &types2.ExecutionPayloadV3{Payload: &types2.ExecutionPayloadV2{Payload: payload, Withdrawals: withdrawals}, ExcessDataGas: excessDataGasReply}, nil
```

Comment:
general question, do we want to combine V2 and an extra field to construct V3? Or would it be better to create an actual version (duplication of fields)?

Reply:
the current style of nesting payloads gets pretty messy once you get past V2 doesn't it?  Not sure what the trade-offs are around diverging from this style so sticking with the prior art for now.